### PR TITLE
feat: group nav by category

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Group the nav into logical categories

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a

## Risk

none

## Testing

n/a

## Screenshots (if applicable)
Screen Shot 2022-08-19 at 9 21 19 AM

### CODE REVIEW

File: src/components/Layout/Header/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -14,26 +16,50 @@ type NavBarProps = {
...
-              href={item.path}
-              to={item.path}
-              label={translate(item.label)}
-              aria-label={translate(item.label)}
-              data-test={`navigation-${item.label.split('.')[1]}-button`}
-            />
-          )
-        })}
+    <Stack width='full' flex='1 1 0%' spacing={6} {...rest}>
+      {groupArray.map((group, id) => {
```
Comment (0xdef1cafe) [inline src/components/Layout/Header/NavBar/NavBar.tsx:30]: memoize plz

File: src/components/Layout/Header/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -14,26 +17,52 @@ type NavBarProps = {
 export const NavBar = ({ isCompact, ...rest }: NavBarProps) => {
   const translate = useTranslate()
   const { routes: pluginRoutes } = usePlugins()
+  const groupColor = useColorModeValue('gray.300', 'gray.600')
+
+  const groupArray = useMemo(() => {
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/NavBar.tsx:22]: ser

Comment (gomesalexandre) [reply src/components/Layout/Header/NavBar/NavBar.tsx:22]: `routeGroups` / `routeChunks` ?

File: src/components/Layout/Header/NavBar/NavBar.tsx [hunk 3]
```diff
@@ -14,26 +17,52 @@ type NavBarProps = {
...
-              label={translate(item.label)}
-              aria-label={translate(item.label)}
-              data-test={`navigation-${item.label.split('.')[1]}-button`}
-            />
-          )
-        })}
+    <Stack width='full' flex='1 1 0%' spacing={6} {...rest}>
+      {groupArray.map((group, id) => {
+        const name = group[0]
+        const values = group[1]
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/NavBar.tsx:35]: Ideally would prefer this to be key-based but if groups are going to be arrays, let's make it:

```suggestion
        const [name, values] = group
```

Or even better, destructure it on the line above

File: src/components/Layout/Header/NavBar/NavBar.tsx [hunk 4]
```diff
@@ -14,26 +17,52 @@ type NavBarProps = {
...
+                color={groupColor}
+                fontSize='xs'
+                textTransform='uppercase'
+                fontWeight='bold'
+                letterSpacing='wider'
+                display={{ base: 'none', '2xl': 'block' }}
+                translation={`navBar.${name}`}
+              />
+            )}
+            {values.map((item: Route, idx: number) => (
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/NavBar.tsx:50]:
```suggestion
            {values.map((item: Route, index: number) => (
```

File: src/components/Layout/Header/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -14,26 +17,52 @@ type NavBarProps = {
 export const NavBar = ({ isCompact, ...rest }: NavBarProps) => {
   const translate = useTranslate()
   const { routes: pluginRoutes } = usePlugins()
+  const groupColor = useColorModeValue('gray.300', 'gray.600')
+
+  const groupArray = useMemo(() => {
+    const allRoutes = union(routes, pluginRoutes).filter(route => !route.disable && !route.hide)
+    const groups = allRoutes.reduce(
+      (entryMap, e) => entryMap.set(e.category, [...(entryMap.get(e.category) || []), e]),
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/NavBar.tsx:25]:
```suggestion
      (entryMap, currentRoute) => entryMap.set(e.category, [...(entryMap.get(currentRoute.category) || []), currentRoute]),
```

or
```suggestion
      (entryMap, route) => entryMap.set(e.category, [...(entryMap.get(route.category) || []), route]),
```

**Review by gomesalexandre — DISMISSED** [review]
Stamping as this is conceptually fine and does what it says on the label, but see comments

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**